### ISSUE TITLE
Bug: Homey app shows "Device unavailable" while the Pulse device is actually available in the Tibber app

### ISSUE THREAD

**Issue by iver56** [tibber/com.tibber.athom#84]
### Homey model

Homey Pro Early 2019

### Firmware

10.0.0

### Power adapter

None

### Tibber version

1.9.18

### The bug

The icon for my Tibber Pulse shows a red triangle icon. When I open it I see header "Device unavailable" and error message "Tibber home with specified id not found. Please re-add device.". It seems like this problem in the Homey app doesn't resolve itself automatically. I can see the Tibber Pulse reporting values in real time in the official Tibber Android app.

I have not changed anything in the Tibber app, I have not changed my Tibber Pulse. I haven't changed my Wi-Fi router or anything like that. I have one home with one Tibber Pulse.

It feels like this is the 3rd or 4th time this happens. I don't want to manually check regularly if I need to re-add my Pulse device because some id might have changed somewhere in an internal system. I want it to _just work_, without manual help. Ideally, the homey app should be able to resolve the id/connection automatically if it got lost somehow. In my situation, with just one Tibber Pulse in my account, that should be doable.

### How frequently does this happen?

On the order of 2-3 times per year

### Tibber Premium

I have an active Tibber subscription

**Comment by iver56** [comment]
This bug still exists. I experienced it a few days ago after my internet service provider did some maintenance that led to a few hours of downtime on the internet connection in the home.

My Homey FW is now 12.0.0, and the Tibber homey app version is still 1.9.18.

If you need more info, feel free to ask.

**Comment by iver56** [comment]
This is _still_ a problem. Manually removing and re-adding the Tibber Pulse regularly is annoying. Please pay attention to this bug, and let me know if there is anything I can do to help.

**Comment by iver56** [comment]
Yeah, mine didn't automatically reconnect either. I had to remove and add it. There is room for improvement.

**Comment by toini** [comment]
I'm suspecting the issue is with [getHomeFeatures()](https://github.com/tibber/com.tibber.athom/blob/c343f713983026405ba744056643637cfbe382bf/lib/tibber-api.ts#L193) returning undefined for [viewer?.home?.features?.realTimeConsumptionEnabled](https://github.com/tibber/com.tibber.athom/blob/c343f713983026405ba744056643637cfbe382bf/drivers/pulse/device.ts#L133) and we end up disabling the the device.

To confirm this, would you be able to find logs before and after `Tibber home with specified id not found. Please re-add device`?  Especially ones that might be saying `Error while fetching home features` originating from this [row](https://github.com/tibber/com.tibber.athom/blob/c343f713983026405ba744056643637cfbe382bf/lib/tibber-api.ts#L205)? @marcelismus @iver56

**Comment by iver56** [comment]
Thanks for looking into this. How/where would I access those logs? Sorry, I did not note the exact time, but it was probably around the same time when other users (like @Marcelismus) also had issues, due to the downtime.

I'll send you my email address

**Comment by iver56** [comment]
Done. I have emailed you the secret id.

**Comment by toini** [comment]
We found an issue with a websocket connection initialization request failure getting wrongfully flagged as failed authentication during high load situation which leads to homey app switching the device to unavailable. We've deployd a fix now along with other remediations to help reduce the likelyhood of reconnects. Can you please re-open this issue if the problem reoccurs?

**Comment by philipostli** [comment]
@toini Happened again today around 8 in the morning. The fix is committed after v1.10.2 which is the most recent in test. Latest production version is 1.9.21. Would you mind creating another version for test?
I made a diagnosis report just now: 5608e278-98eb-4302-84b7-7841a90d2612
